### DISCUSSION TITLE
about dnscrypt's windows service

### DISCUSSION THREAD

**Discussion by galpt** [DNSCrypt/dnscrypt-proxy#1849] [Q&A]
I saw inside the `dnscrypt-proxy/windows` that [service-restart.bat](https://github.com/DNSCrypt/dnscrypt-proxy/blob/master/windows/service-restart.bat) is using `dnscrypt-proxy.exe -service stop` and `dnscrypt-proxy.exe -service start` instead of `dnscrypt-proxy.exe -service restart`.
If this is expected, what's the difference if we just use `dnscrypt-proxy.exe -service restart`?

**Comment by jedisct1** [comment]
`-service restart` restarts the service after a configuration change; i.e. the service should be already running.
